traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/util/Fetcher.java

```diff
@@ -233,4 +239,23 @@ public void createStringBuilderFromResponse (final StringBuilder sb, final HttpU
 			}
 		}
 	}
+
+	private void connectionFailed(final HttpURLConnection http, final String url) {
+		String httpUrl = url;
+		String responseCode = "(none)";
+		try {
+			httpUrl = http.getURL().toString();
+			responseCode = String.valueOf(http.getResponseCode());
+		} catch (Exception e2) {
+			// Don't care
+			LOGGER.info("Exception during call attempt to retrieve url or responseCode from http");
```

Comment:
Should this be `error` instead of `info`? Also, should we log the exception?

Reply:
We can definitely log the exception - I'll add that. I'll also add the exception to the main "Failed Http Request to "... error message.

Regarding this being an info vs error level. I'm of the opinion that generating an error message here (second/embedded catch block) would confuse diagnosis and debugging of the actual underlying issue that landed us in the main catch-block - which is likely to be related to a connection failure. If we generated an error while trying to generate the error message we'd just be adding noise to the problem space.

Reply:
Ok, maybe `debug` rather than `info`?

Reply:
I like that ... makes sense